Code-the-Dream-School/ee-prac-team3-front
# Feature/logout modal/code refactor

## Purpose:
To fix bug with logout, add modal on logout, and simplify code

## Summary:
fix bug in logout (userAuth context was not being updated)
add logout modal
refactored code to create generic API request function
refactored code to add exportFunctions component with functions that are used in the App component to simplify App code.

## Front End Change:
App.jsx
Login.jsx
Signup.jsx
exportFunctions.js


## Test Steps:
Can test refactored API call by logging in with email: "bear@email.com" and password: "1234", user should be logged in and redirected to the home page where the "Quizzes" and "About" buttons should be visible.  Reviewer can test signup by creating a new account and using the credentials to Login.  Reviewer can test "Logout" changes by navigating to the "Quizzes" page, clicking on the User Profile and pressing Logout.
